pepper-apis/src/main/java/org/broadinstitute/ddp/model/event/EventConfiguration.java

```diff
@@ -77,46 +85,51 @@ public EventConfiguration(EventConfigurationDto dto) {
-                break;
-            case MARK_ACTIVITIES_READ_ONLY:
-                eventAction = new MarkActivitiesReadOnlyEventAction(this, dto);
-                break;
-            case PDF_GENERATION:
-                eventAction = new PdfGenerationEventAction(this, dto);
-                break;
-            case REVOKE_PROXIES:
-                eventAction = new RevokeProxiesEventAction(this, dto);
-                break;
-            case UPDATE_USER_STATUS:
-                eventAction = new UpdateUserStatusEventAction(this, dto);
-                break;
-            default:
-                throw new DDPException("Event action type: " + eventActionType.name() + " is not properly configured in "
-                        + "the EventConfiguration ctor");
-
+        try {
+            switch (eventActionType) {
+                case ACTIVITY_INSTANCE_CREATION:
+                    eventAction = new ActivityInstanceCreationEventAction(this, dto);
+                    break;
+                case NOTIFICATION:
+                    eventAction = new NotificationEventAction(this, dto);
+                    break;
+                case USER_ENROLLED:
+                    eventAction = new EnrollUserEventAction(this, dto);
+                    break;
+                case ANNOUNCEMENT:
+                    eventAction = new AnnouncementEventAction(this, dto);
+                    break;
+                case CREATE_INVITATION:
+                    eventAction = new CreateInvitationEventAction(this, dto);
+                    break;
+                case COPY_ANSWER:
+                    eventAction = new CopyAnswerEventAction(this, dto);
+                    break;
+                case HIDE_ACTIVITIES:
+                    eventAction = new HideActivitiesEventAction(this, dto);
+                    break;
+                case MARK_ACTIVITIES_READ_ONLY:
+                    eventAction = new MarkActivitiesReadOnlyEventAction(this, dto);
+                    break;
+                case PDF_GENERATION:
+                    eventAction = new PdfGenerationEventAction(this, dto);
+                    break;
+                case REVOKE_PROXIES:
+                    eventAction = new RevokeProxiesEventAction(this, dto);
+                    break;
+                case UPDATE_USER_STATUS:
+                    eventAction = new UpdateUserStatusEventAction(this, dto);
+                    break;
+                case UPDATE_CUSTOM_WORKFLOW:
+                    eventAction = new UpdateCustomWorkflowEventAction(this, dto, new EventActionPubSubPublisher());
+                    break;
+                default:
+                    throw new DDPException("Event action type: " + eventActionType.name() + " is not properly configured in "
+                            + "the EventConfiguration ctor");
+            }
+        } catch (Exception e) {
+            LOG.error(format("Error handle event %s", eventActionType), e);
```

Comment:
This will swallow the exception... meaning the constructor will finish as if nothing has happened. So should forego catching the exception so it propagates up and halts the execution? Should we rethrow the exception here?

Reply:
But if an exception not swallowed and re-thrown then it can interrupt the execution of other logic where events thrown. Is it good that due to some PubSub communication problem a process which triggers an event will be failed? May be for reliable delivering of an event to DSM via PubSub it needs to invent some special approach: some flag in table `update_custom_workflow_event_action` which should be checked by an async job publishing events to PubSub: and if flag is false then try to publish event and set it on successful  publishing ...

Reply:
I fixed: DDPException() is re-thrown now, but I'm not sure that it is a good solution. In future I suggest to implement asynchronous job for such events publishing.